src/index/VocabularyGeneratorImpl.h

```diff
@@ -224,43 +222,43 @@ void VocabularyMerger::doActualWrite(
 }
 
 // ______________________________________________________________________________________________
+template <class Pred>
 void writePartialIdMapToBinaryFileForMerging(
-    std::shared_ptr<const ad_utility::HashMap<string, Id>> map,
-    const string& fileName, StringSortComparator comp,
-    const bool doParallelSort) {
+    std::shared_ptr<const Index::ItemMap> map, const string& fileName,
+    Pred pred) {
   LOG(INFO) << "Creating partial vocabulary from set ...\n";
   std::vector<std::pair<string, Id>> els;
   els.reserve(map->size());
   els.insert(begin(els), begin(*map), end(*map));
   LOG(INFO) << "... sorting ...\n";
 
-  auto pred = [comp](const auto& p1, const auto& p2) {
-    return comp(p1.first, p2.first);
-  };
-  if constexpr (USE_PARALLEL_SORT) {
-    if (doParallelSort) {
+  auto sort = [&els](const auto& pred) {
+    if constexpr (USE_PARALLEL_SORT) {
       __gnu_parallel::sort(begin(els), end(els), pred,
                            __gnu_parallel::parallel_tag(NUM_SORT_THREADS));
     } else {
       std::sort(begin(els), end(els), pred);
     }
-  } else {
-    std::sort(begin(els), end(els), pred);
-    (void)doParallelSort;  // avoid compiler warning for unused value.
-  }
-  LOG(INFO) << "Done creating vocabulary.\n";
+  };
+
+  auto comp = [&pred](const auto& a, const auto& b) {
```

Comment:
This can be constr, right oreven `constexpr` maybe then the compiler has a chance of optimizing this a bit more?